### PR TITLE
fixed up readme and added near-shell to dockerfile

### PR DESCRIPTION
**Pull request by willemneal**
Also moved context.json to test folder to not clutter root directory

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,14 +8,59 @@ Contract in `assembly/main.ts` provides methods to increment / decrement counter
 Plus and minus buttons increase and decrease value correspondingly. When button L is toggled, counter will add or minus 10 a time. RS button is for reset. LE and RE buttons to let the robot wink to you.
 
 ## To Run
+[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/nearprotocol/counter)
+
+If running in gitpod you can skip to [login](#login).
```
Comment (icerove) [inline README.md]: suggest to  make Gitpod all in one format to make it clear it is all that Gitpod. also suggest to say `skip setup`

File: README.md [hunk 2]
```diff
@@ -9,14 +8,59 @@ Contract in `assembly/main.ts` provides methods to increment / decrement counter
...
 ```
 yarn start
 ```
 
 ## To Test
 
 ```
-yarn test
+yarn asp # as-pect tests
+yarn jest # jest tests
```
Comment (icerove) [inline README.md]: forgot to say that the jest test command should `NODE_ENV=ci` we use this environment variable to test on ci

Comment (willemneal) [reply README.md]: Okay.  Should I add that to the gitpod environment too?

**Comment by mikedotexe** [comment]
I think we'll want the quickest process to be:
`yarn`
`yarn dev`

I believe with this branch you need a `yarn build` in the middle. If we could add that to `package.json` that'd be awesome. ❤️

File: package.json [hunk 3]
```diff
@@ -7,11 +7,17 @@
     "deploy:contract": "near deploy",
     "deploy:pages": "gh-pages -d src",
     "deploy": "yarn build && yarn deploy:contract && yarn deploy:pages",
+    "deploy:dev": "yarn build && near dev-deploy",
+    "deploy:dev:init" : "near dev-deploy --init",
     "prestart": "yarn build && yarn deploy:contract",
+    "dev": "yarn deploy:dev:init && yarn deploy:dev && yarn watch",
```
Comment (willemneal) [inline package.json]: >I think we'll want the quickest process to be:
>yarn
>yarn dev

>I believe with this branch you need a yarn build in the middle. If we could add that to package.json that'd be awesome. ❤️
@mikedotexe Do you mean here?

`dev:deploy` task has a `yarn build`

Comment (mikedotexe) [reply package.json]: ah yeah you're right

File: package.json [hunk 4]
```diff
@@ -7,11 +7,17 @@
     "deploy:contract": "near deploy",
     "deploy:pages": "gh-pages -d src",
     "deploy": "yarn build && yarn deploy:contract && yarn deploy:pages",
+    "deploy:dev": "yarn build && near dev-deploy",
+    "deploy:dev:init" : "near dev-deploy --init",
```
Comment (mikedotexe) [inline package.json]: Looks like we're dev-deploying twice when the user runs `yarn dev`
Screenshot 2020-03-10 10 11 20

Comment (willemneal) [reply package.json]: So I guess we can just add `--init` when doing the last deploy and get rid of the init task?

Comment (mikedotexe) [reply package.json]: as we talked about offline, sounds like we want to get rid of that `--init` call I believe

**Comment by mikedotexe** [comment]
I was trying to see if we've made progress on the initial gas for the access key.
I ran `yarn dev` and think that parcel is not getting the proper dev-deploy id as shown here:
Screenshot 2020-03-10 14 45 42

I think this is similar to why I used the `cat` approach here:
https://github.com/nearprotocol/token-contract-as/blob/master/package.json#L7

**Comment by mikedotexe** [comment]
Looks like after using `near login` and changing the top line of `src/config.js` and then `yarn start` I am not getting the counter to load either. Nondescript console errors this time, just pointing it out. Seems like it'd be good to standardize to use parcel for both `yarn dev` and `yarn start` if we can, probably removing `serve`. That might be the underlying issue 🤷‍♂

**Pull request merged**